### PR TITLE
feat(personalization-utils): add trimHtml for dynamic container…

### PR DESCRIPTION
**Pull request by teleaziz**
…s processing on the edge`

## Add trimHtml function for dynamic personalization containers

This PR introduces the `getTrimmedHtml` function, a new utility for processing dynamic personalization containers in HTML content at the edge or in server-side rendering scenarios.
- Enables edge-side and server-side processing of personalized content
- Allows for caching or static generation of personalized pages

### CODE REVIEW

**Comment by steve8708** [comment]
Hell yes! Wdyt of just calling it trimHtml()!

File: packages/personalization-utils/README.md [hunk 1]
```diff
@@ -65,12 +65,48 @@ export default function middleware(request) {
...
-  setCookie(`builder.userAttributes.audience`, audience)
+```typescript
+const audience = await myCDP.identifyAudience(userID);
+// this will include the `audience` in all api calls and save it in a cookie `builder.userAttributes`
+builder.setUserAttributes({ audience })
 ```
+
 Once the cookie is set, all builder content matching from now on will weigh in the current audience segment.
 
+## Using trimHtml for Dynamic Containers
```
Comment (mrkoreye) [inline packages/personalization-utils/README.md:76]: great docs 👍

File: packages/personalization-utils/src/get-trimmed-html.test.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
+import { trimHtml } from './utils'; // Update this import to match your file structure
+
+describe('trimHtml', () => {
```
Comment (mrkoreye) [inline packages/personalization-utils/src/get-trimmed-html.test.ts:3]: yes test!

**Review by mrkoreye — APPROVED** [review]
this is a great idea

**Pull request merged**